src/libsyntax/ast.rs

```diff
@@ -451,6 +451,15 @@ pub struct Crate {
     pub module: Mod,
     pub attrs: Vec<Attribute>,
     pub span: Span,
+    pub proc_macros: Vec<ProcMacroInfo>
+}
+
+/// Data from any proc macro declarations
+/// we collected from this crate
+#[derive(Clone, RustcEncodable, RustcDecodable, Debug)]
+pub struct ProcMacroInfo {
+    pub span: Span,
+    pub attrs: Vec<Attribute>
```

Comment:
@petrochenkov This is the part I was r?-ing you for but it's simpler than I initially thought.